### PR TITLE
Port Maybe's Generic instance to this repo

### PR DESCRIPTION
**Pull request by JordanMartinez**

### CODE REVIEW

File: src/Data/Maybe.purs [hunk 1]
```diff
@@ -216,6 +217,16 @@ instance showMaybe :: Show a => Show (Maybe a) where
...
+instance genericMaybe
+  :: Generic (Maybe a) (Sum (Constructor "Nothing" NoArguments)
+                            (Constructor "Just" (Argument a))) where
+  to (Inl _) = Nothing
+  to (Inr (Constructor (Argument a))) = Just a
+
+  from Nothing = Inl (Constructor NoArguments)
+  from (Just a) = Inr (Constructor (Argument a))
+
+
```
Comment (kl0tl) [inline src/Data/Maybe.purs:229]: Can’t this instance be derived by the compiler?

Comment (JordanMartinez) [reply src/Data/Maybe.purs:229]: Probably.

Comment (garyb) [reply src/Data/Maybe.purs]: That was one of the issues having it defined in `Generic` actually; it couldn't derive it there :smile:

**Comment by JordanMartinez** [comment]
Huh... even when I compile this locally, the code still fails to compile with this error:
```
[1/1 UnknownImport] src/Data/Maybe.purs:13:26

  13  import Data.Generic.Rep (class GenericRep)
                               ^^^^^^^^^^^^^^^^
  
  Cannot import type class GenericRep from module Data.Generic.Rep
  It either does not exist or the module does not export it.

           Src   Lib   All
Warnings   0     0/1   0/1
Errors     1     0     1  
```
However, the module exists and the `Generic` class is exported in prelude now: https://github.com/purescript/purescript-prelude/blob/master/src/Data/Generic/Rep.purs#L2

**Review by kl0tl — APPROVED** [review]

**Pull request merged**